integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/mutation/merge/CarbonMergeDataSetCommand.scala

```diff
@@ -0,0 +1,520 @@
+
+    val executorErrors = ExecutionErrors(FailureCauses.NONE, "")
+    val trxMgr = TranxManager(model.getFactTimeStamp)
+
+    val mutationAction = MutationActionFactory.getMutationAction(sparkSession,
+      carbonTable, hasDelAction, hasUpdateAction,
+      insertHistOfUpdate, insertHistOfDelete)
+
+    val tuple = mutationAction.handleAction(processedRDD, executorErrors, trxMgr)
+
+    // In case user only has insert action.
+    if (!(hasDelAction || hasUpdateAction)) {
+      processedRDD.count()
+    }
+    LOGGER.info(s"Total inserted rows: ${stats.insertedRows.sum}")
+    LOGGER.info(s"Total updated rows: ${stats.updatedRows.sum}")
+    LOGGER.info(s"Total deleted rows: ${stats.deletedRows.sum}")
+    LOGGER.info(
+      " Time taken to merge data  : " + tuple + " :: " + (System.currentTimeMillis() - st))
+
+    val segment = new Segment(model.getSegmentId,
+      SegmentFileStore.genSegmentFileName(
+        model.getSegmentId,
+        System.nanoTime().toString) + CarbonTablePath.SEGMENT_EXT,
+      CarbonTablePath.getSegmentPath(carbonTable.getTablePath,
+        model.getSegmentId), Map.empty[String, String].asJava)
+    val writeSegment =
+      SegmentFileStore.writeSegmentFile(carbonTable, segment)
+
+    if (writeSegment) {
+      SegmentFileStore.updateTableStatusFile(
+        carbonTable,
+        model.getSegmentId,
+        segment.getSegmentFileName,
+        carbonTable.getCarbonTableIdentifier.getTableId,
+        new SegmentFileStore(carbonTable.getTablePath, segment.getSegmentFileName),
+        SegmentStatus.SUCCESS)
+    } else {
+      CarbonLoaderUtil.updateTableStatusForFailure(model)
+    }
+
+    if (hasDelAction || hasUpdateAction) {
+      if (CarbonUpdateUtil.updateSegmentStatus(tuple._1, carbonTable,
+        trxMgr.getLatestTrx.toString, false) &&
+          CarbonUpdateUtil
+            .updateTableMetadataStatus(
+              model.getLoadMetadataDetails.asScala.map(l =>
+                new Segment(l.getMergedLoadName,
+                  l.getSegmentFile)).toSet.asJava,
+              carbonTable,
+              trxMgr.getLatestTrx.toString,
+              true,
+              tuple._2.asJava)) {
+        LOGGER.info(s"Merge data operation is successful for " +
+                    s"${ carbonTable.getDatabaseName }.${ carbonTable.getTableName }")
+      } else {
+        throw new CarbonMergeDataSetException("Saving update status or table status failed")
+      }
+    }
+    // Load the history table if the inserthistorytable action is added by user.
+    HistoryTableLoadHelper.loadHistoryTable(sparkSession, rltn.head, carbonTable,
+      trxMgr, mutationAction, mergeMatches)
+    Seq.empty
+  }
+
+  private def processIUD(sparkSession: SparkSession,
+      frame: DataFrame,
+      carbonTable: CarbonTable,
+      model: CarbonLoadModel,
+      projections: Seq[Seq[MergeProjection]],
+      stats: Stats) = {
+    val conf = SparkSQLUtil.sessionState(sparkSession).newHadoopConf()
```

Comment:
It will be better if you can describe the processing flow briefly in the top of this function

Reply:
ok